src/toil/cwl/cwltoil.py

```diff
@@ -2197,7 +2197,7 @@ def flat_crossproduct_scatter(
                     tool=self.step.embedded_tool,
                     jobobj=updated_joborder,
                     runtime_context=self.runtime_context,
-                    parent_name=f"{self.parent_name}[{n}]",
+                    parent_name=f"{self.parent_name}.{n}]",
```

Comment:
`]` looks like it may have been left here accidentally.

Reply:
Nice catch! Fixed.

Reply:
Good catch.  Fixed!